### ISSUE TITLE
Exception on adding of S4_E882D22F resource.

### ISSUE THREAD

**Issue by KhArtNJava** [Kuree/Sims4Tools#37]
I'm trying to add interaction S4_E882D22F.
Steps:
File - New
Resource - Add
Type = _XML 0xE882D22F
Group = 0
Name = MyFirstInteraction_Test
Press FNV64 Button - OK
Will be rised Exception 
```
Подробная информация об использовании оперативной 
(JIT) отладки вместо данного диалогового 
окна содержится в конце этого сообщения.

************** Текст исключения **************
System.ArgumentOutOfRangeException: Индекс за пределами диапазона. Индекс должен быть положительным числом, а его размер не должен превышать размер коллекции.
Имя параметра: index
   в System.Collections.Generic.List`1.get_Item(Int32 index)
   в S4PIDemoFE.BrowserWidget.MergeNamemap(IEnumerable`1 newMap, Boolean create, Boolean replace)
   в S4PIDemoFE.BrowserWidget.ResourceName(UInt64 instance, String resourceName, Boolean create, Boolean replace)
   в S4PIDemoFE.MainForm.resourceAdd()
   в S4PIDemoFE.MainForm.menuBarWidget1_MBResource_Click(Object sender, MBClickEventArgs mn)
   в S4PIDemoFE.MenuBarWidget.OnMBResource_Click(Object sender, MB mn)
   в S4PIDemoFE.MenuBarWidget.tsMBR_Click(Object sender, EventArgs e)
   в System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   в System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   в System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   в System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   в System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   в System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   в System.Windows.Forms.Control.WndProc(Message& m)
   в System.Windows.Forms.ToolStrip.WndProc(Message& m)
   в System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   в System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Загруженные сборки **************
mscorlib
    Версия сборки: 4.0.0.0
    Версия Win32: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
s4pe
    Версия сборки: 1409.20.48.34124
    Версия Win32: 1409.20.48.34124
    CodeBase: file:///C:/Art/Projects/sims3tools/s4pe.0.3.b/s4pe.exe
----------------------------------------
System.Windows.Forms
    Версия сборки: 4.0.0.0
    Версия Win32: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Версия сборки: 4.0.0.0
    Версия Win32: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System
    Версия сборки: 4.0.0.0
    Версия Win32: 4.0.30319.34239 built by: FX452RTMGDR
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Configuration
    Версия сборки: 4.0.0.0
    Версия Win32: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Custom
    Версия сборки: 1412.28.57.34110
    Версия Win32: 1412.28.57.34110
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/System.Custom.DLL
----------------------------------------
System.Core
    Версия сборки: 4.0.0.0
    Версия Win32: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Версия сборки: 4.0.0.0
    Версия Win32: 4.0.30319.34230 built by: FX452RTMGDR
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
s4pi.CustomForms
    Версия сборки: 1412.28.57.34110
    Версия Win32: 1412.28.57.34110
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.CustomForms.DLL
----------------------------------------
s4pi.Interfaces
    Версия сборки: 1412.28.57.34110
    Версия Win32: 1412.28.57.34110
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.Interfaces.DLL
----------------------------------------
s4pi.WrapperDealer
    Версия сборки: 1412.28.57.34111
    Версия Win32: 1412.28.57.34111
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.WrapperDealer.DLL
----------------------------------------
DDSPanel
    Версия сборки: 1412.28.57.34112
    Версия Win32: 1412.28.57.34112
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/DDSPanel.DLL
----------------------------------------
s4pi.AnimationResources
    Версия сборки: 0.0.0.0
    Версия Win32: 0.0.0.0
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.AnimationResources.DLL
----------------------------------------
s4pi.GenericRCOLResource
    Версия сборки: 1412.28.57.34111
    Версия Win32: 1412.28.57.34111
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.GenericRCOLResource.DLL
----------------------------------------
s4pi.CASPartResource
    Версия сборки: 1412.28.57.34112
    Версия Win32: 1412.28.57.34112
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.CASPartResource.DLL
----------------------------------------
s4pi.Settings
    Версия сборки: 1412.28.57.34110
    Версия Win32: 1412.28.57.34110
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.Settings.DLL
----------------------------------------
s4pi.CatalogResource
    Версия сборки: 1412.28.57.34111
    Версия Win32: 1412.28.57.34111
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.CatalogResource.DLL
----------------------------------------
s4pi.Controls
    Версия сборки: 1412.28.57.34112
    Версия Win32: 1412.28.57.34112
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.Controls.DLL
----------------------------------------
s4pi.DataResource
    Версия сборки: 1412.28.57.34112
    Версия Win32: 1412.28.57.34112
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.DataResource.DLL
----------------------------------------
s4pi.DefaultResource
    Версия сборки: 1412.28.57.34111
    Версия Win32: 1412.28.57.34111
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.DefaultResource.DLL
----------------------------------------
s4pi.Extensions
    Версия сборки: 1412.28.57.34111
    Версия Win32: 1412.28.57.34111
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.Extensions.DLL
----------------------------------------
s4pi.Filetable
    Версия сборки: 1412.28.57.34112
    Версия Win32: 1412.28.57.34112
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.Filetable.DLL
----------------------------------------
s4pi.Helpers
    Версия сборки: 1412.28.57.34112
    Версия Win32: 1412.28.57.34112
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.Helpers.DLL
----------------------------------------
s4pi.ImageResource
    Версия сборки: 1412.28.57.34111
    Версия Win32: 1412.28.57.34111
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.ImageResource.DLL
----------------------------------------
s4pi.MeshChunks
    Версия сборки: 1412.28.57.34112
    Версия Win32: 1412.28.57.34112
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.MeshChunks.DLL
----------------------------------------
s4pi.RCOLChunks
    Версия сборки: 1412.28.57.34111
    Версия Win32: 1412.28.57.34111
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.RCOLChunks.DLL
----------------------------------------
s4pi.MiscellaneousResources
    Версия сборки: 1412.28.57.34112
    Версия Win32: 1412.28.57.34112
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.MiscellaneousResources.DLL
----------------------------------------
s4pi.Package
    Версия сборки: 1412.28.57.34111
    Версия Win32: 1412.28.57.34111
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.Package.DLL
----------------------------------------
s4pi.RigResource
    Версия сборки: 1412.28.57.34112
    Версия Win32: 1412.28.57.34112
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.RigResource.DLL
----------------------------------------
s4pi.StblResource
    Версия сборки: 1412.28.57.34111
    Версия Win32: 1412.28.57.34111
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.StblResource.DLL
----------------------------------------
s4pi.TextResource
    Версия сборки: 1412.28.57.34112
    Версия Win32: 1412.28.57.34112
    CodeBase: file:///c:/Art/Projects/sims3tools/s4pe.0.3.b/s4pi.TextResource.DLL
----------------------------------------
mscorlib.resources
    Версия сборки: 4.0.0.0
    Версия Win32: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/mscorlib.resources/v4.0_4.0.0.0_ru_b77a5c561934e089/mscorlib.resources.dll
----------------------------------------
Microsoft.GeneratedCode
    Версия сборки: 1.0.0.0
    Версия Win32: 4.0.30319.34230 built by: FX452RTMGDR
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Windows.Forms.resources
    Версия сборки: 4.0.0.0
    Версия Win32: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.resources/v4.0_4.0.0.0_ru_b77a5c561934e089/System.Windows.Forms.resources.dll
----------------------------------------

************** Оперативная отладка (JIT) **************
Для подключения оперативной (JIT) отладки файл .config данного
приложения или компьютера (machine.config) должен иметь
значение jitDebugging, установленное в секции system.windows.forms.
Приложение также должно быть скомпилировано с включенной
отладкой.

Например:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

При включенной отладке JIT любое необрабатываемое исключение
пересылается отладчику JIT, зарегистрированному на данном компьютере,
вместо того чтобы обрабатываться данным диалоговым окном.

```

**Comment by Kuree** [comment]
Thank you for your detailed bug report. I'm able to reproduce the bug here. I'll fix it as soon as possible.

**Comment by Kuree** [comment]
Bug found. The fix(cc87787e352698832cc7c344b93cc441c4d4545e) will be applied to next release.
